### ISSUE TITLE
UWP App Crash while running in UI thread

### ISSUE THREAD

**Issue by shishir42** [Redth/ZXing.Net.Mobile#557]
App is crashing in Windows 10 (UWP) app. 

I am using ZXing.Net.Mobile in UWP app and Invoking scanner object inside UI thread.

Below code :

private async Task<ZXing.Result> ScanBarcode()
        {
            ZXing.Result result = null;
            var taskCompletionSource = new TaskCompletionSource<ZXing.Result>();
            await Windows.ApplicationModel.Core.CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(
                              Windows.UI.Core.CoreDispatcherPriority.Normal,
                              async () =>
                              {
                                  var scanner = new MobileBarcodeScanner();
                                  var scannerOption = new MobileBarcodeScanningOptions { TryHarder = true, AutoRotate = false };
                                  scanner.TopText = "Focus the barcode or QR code in this frame. Wait until it is automatically scanned.";
                                  ////this.Log(LogLevel.Info, null, "Barcode scanner object created.");

                                  scanner.UseCustomOverlay = false;
                                  ////this.CheckDeviceCameraFlash();

                                  // Use front camera for Desktops/Tablets
                                  if (AnalyticsInfo.VersionInfo.DeviceFamily == "Windows.Desktop")
                                  {
                                      scannerOption.UseFrontCameraIfAvailable = true;
                                  }
                                  result = await scanner.Scan(scannerOption);
                                  taskCompletionSource.SetResult(result);
                                  scanner.Cancel();
                              });

            result = await taskCompletionSource.Task;
            return result;
        }

But this method i am calling from non ui thread.
Once the execution is finished and passing value in UI that time i am getting crash.
As per my understanding, once i invoking scanner object in UI thread and after reading data the UI thread is still alive because of that other part of view is not alive.

Can you please suggest that how i will detached Windows.ApplicationModel.Core.CoreApplication.MainView.CoreWindow.Dispatcher

and return back to my current view.

My app is using webview control and called is invoking from html ui control.

Crash Log Info -- 

	{"Exception from HRESULT: 0x80020101"}
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Intelatrac.Mobile.RTComponent.UWP.NavigatorRT.<>c__DisplayClass11_0.<<Navigator_OnVMPropertyChanged>b__0>d.MoveNext()

Please suggest on this.

**Comment by Redth** [comment]
Thanks for reporting this issue!  Unforunately it took me way too long to respond 😭.  Sorry, I apologize!  Recently the source code for this project was completely refactored to modernize it.  Many PR's were included in this effort, and many bugs were hopefully fixed.  Please try out the latest 3.x series of NuGet packages (currently in prerelease).  To try and make the project more maintainable in my spare time going forward, I've decided to close all existing issues to start with a clean slate.  If you're still experiencing this issue on the newest version, please open a new issue with as much detail as possible.  Thank you for your patience and understanding!  Happy scanning!
